Gems/PhysX/Code/Tests/EditorTestUtilities.cpp

```diff
@@ -85,6 +85,15 @@ namespace PhysXEditorTests
             AZ::NonUniformScaleRequestBus::Event(editorEntity->GetId(), &AZ::NonUniformScaleRequests::SetScale, *nonUniformScale);
         }
 
+        // If using a dynamic rigid body then reactivate the entity to recreate the editor world body.
+        // This is because the dynamic rigid body only applies the changes in the next simulation tick,
+        // which happens automatically in the editor but not in test environment.
+        if (rigidBodyType == RigidBodyType::Dynamic)
+        {
+            editorEntity->Deactivate();
+            editorEntity->Activate();
+        }
```

Comment:
suggestion: Should we move this and the comment to a function to reduce the duplication below? Can this just be a member function?

Reply:
> Yeah, I was just thinking the same thing. There's quite a lot of duplication in general between these functions. It could be nice to have them do something like
>
> SetupGenericStuff();
// shape specific logic
...
DoGenericFinalization();

I'm happy to continue refactoring, but I think the granularity at the moment of these `CreateXXXColliderEditorEntity` functions is quite nice because it's very easy to read the steps of how to setup a collider. It is the same pattern, but what they do is different when creating the entity and configuring the components, so refactoring beyond this point I think it's when it'll start to obfuscate the coded rather than helping. Yes, they'll be less lines of code, but for people looking for code snippets or examples of how to do create colliders it'll just make it more complicated to follow.

If you feel strongly I should give it another pass to refactor all these helpers let me know.

Cheers

Reply:
>I'm also wondering if there's a slightly less clunky was to trigger the editor body recreation. I was wondering if it would be OK to make either CreateEditorWorldRigidBody or OnConfigurationChanged public? I guess you'd still need a comment to explain why that was getting called though.

The `OnColliderChanged` is public and it's called by the collider, but the issue is the rigid body just caches a boolean indicating it'll recreate itself in the next simulation tick. I guess this was done because there are many notifications that could trigger recreate the rigid body and many can happen in 1 frame, so it'd be very slow to recreate the body for each notification.

This could be done instead to make the rigid body recreate the body:

````
        // Update editor world to let updates to be applied
        auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
        physicsSystem->Simulate(0.1f);
````

But I didn't like this approach because these are helpers to create the entity, so trigger simulation in the middle of the "setup" step of an unit tests seems odd.

It might be better to add a function to a bus to force the recreation of the rigid body and call it. But this is now out of scope of this PR I'd say.

Reply:
I was thinking about OnConfigurationChanged rather than OnColliderChanged, but it's not important.

These are all just suggestions, the PR is perfectly fine with or without them.

Reply:
Maybe only have the massive comment once then?

Reply:
It would be trivial to make this a free function you call from all the places you've added it

Reply:
I'll move it to a function.

Reply:
Thank you! ❤️